vue-app/src/api/recipient-registry-optimistic.ts

```diff
@@ -512,4 +520,39 @@ export async function registerProject(
   return transaction
 }
 
+export async function rejectProject(
+  registryAddress: string,
+  recipientId: string,
+  recipientAddress: string,
+  signer: Signer
+) {
+  const registry = new Contract(
+    registryAddress,
+    OptimisticRecipientRegistry,
+    signer
+  )
+  const transaction = await registry.challengeRequest(
+    recipientId,
+    recipientAddress
```

Comment:
- [ ] This is not correct, we should use the `request.requester` address here. Will see how to obtain that address.

Reply:
@wackerow as we agreed, since the `challengeRequest` function expects a beneficiary address:
```
function challengeRequest(bytes32 _recipientId, address payable _beneficiary) {
	...
	bool isSent = _beneficiary.send(request.deposit);
	...
}
```
we said that we should return the base deposit to the requester.
 
That is fine. The problem comes in how we read from the consumer's perspective (vue app) that requester address.
 
 I think there are two options:
 1. To add the requester address on the `RequestSubmitted` event.
 2. Or to change the `challengeRequest` function to send it by default to the `request.requester` address.
 
Not sure which one to implement. In both, we will need to modify something in the contract. Perhaps the first option has less impact since we are adding something. In the second one, we modify a bit the behavior.

cc @samajammin @corwintines

Reply:
@pettinarip I think updating the `RequestSubmitted` event (# 1) to include `msg.sender` seems most appropriate. Then we should be able to use that to make sure the deposit is returned to the right address.

Reply:
Done!